xchange-stream-bitmex/src/main/java/info/bitrich/xchangestream/bitmex/BitmexStreamingMarketDataService.java

```diff
@@ -48,33 +53,72 @@ private String getBitmexSymbol(CurrencyPair currencyPair) {
   @Override
   public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
     String instrument = getBitmexSymbol(currencyPair);
-    String channelName =
-        args != null && args.length > 0
-            ? String.format("orderBookL2_25:%s", instrument)
-            : String.format("orderBookL2:%s", instrument);
-
-    return streamingService
+    String channelName = String.format("orderBookL2:%s", instrument);
+    boolean fullBook = false;
+    if (args != null && args.length > 0) {
+      if (args[0] instanceof String && "10".equals(args[0])) {
+        channelName = String.format("orderBook10:%s", instrument);
+        fullBook = true;
+      } else {
+        channelName = String.format("orderBookL2_25:%s", instrument);
+      }
+    }
+
+    if (fullBook) {
+      return streamingService
+        .subscribeBitmexChannel(channelName)
+        .map(s -> {
+          RawOrderBook orderBook = s.toRawOrderBook();
+          if (orderBook != null) {
+            List<LimitOrder> asks = new ArrayList<>(10);
```

Comment:
this might actually be worse, given the default is 16, why don't you set it to orderBook.getAsks().size() and orderBook.getBids().size() ?

Reply:
fixed, my fault (actually it will be 10  due to channel description: "orderBook10")